src/EntityFactory.ts

```diff
@@ -81,15 +78,19 @@ export class EntityFactory {
 	 * @param {Class}  Type      Type of entity to instantiate
 	 * @return {type}
 	 */
-	createByType(
+	createByType<T extends typeof Room | typeof Npc | typeof Item>(
 		area: Area,
 		entityRef: EntityReference,
-		Type: typeof Room | typeof Npc | typeof Item
+		Type: T
 	) {
 		const definition = this.getDefinition(entityRef);
 		if (!definition) {
-			throw new Error(`[${Type.name}Factory] No Entity definition found for ${entityRef}`);
+			throw new Error(
+				`[${Type.name}Factory] No Entity definition found for ${entityRef}`
+			);
 		}
+
+		// @ts-ignore
```

Comment:
So here's a spot where I couldn't quite work out the proper typing. Probably since this is pretty sketch way to do something. Essentially the problem is it can't match the 'Type' and 'definition' to being of the same class when instantiating it. Not sure if we want to have the @ts-ignore, but everything was 'any' before so....